app/controllers/teams_controller.rb

```diff
@@ -87,7 +89,12 @@ def team_editable?
   end
 
   def team_params
-    params.require(:team).permit(:team_name, :affiliation, :division_id, user_invites_attributes: [:email])
+    if @game.location_required?
```

Comment:
No need to complicate the parameter logic here, parameter sanitation is mainly to protect against abuse of fields (such as stopping random users from being able to mark themselves as team_captain for example). Since the worst thing someone can do with the team_location field is set the location, there is no reason to complicate the parameter permitting here.